src/common/veteran.js

```diff
@@ -675,6 +675,22 @@ function veteranToApplication(veteran) {
       case 'deductibleMedicalExpenses':
         return Number(value.value);
 
+      // Optional Suffix fields
+      case 'veteranFullName':
+        if (value.suffix.value === '') { return undefined; }
+        break;
+
+      // Optional Date fields
+      case 'spouseDateOfBirth':
+      case 'dateOfMarriage':
+      case 'medicarePartAEffectiveDate':
+        if (value.day.value === '') { return undefined; }
+        break;
+
+      // Optional SSN fields
```

Comment:
You can combine the Optional Suffix fields and Optional SSN fields.

The grouping here makes the most sense to be if they are done by "type" instead of semantic value.  Basically, we have things that look like dates, and things that look like strings.